README.md

```diff
@@ -15,11 +30,51 @@ Current status:
 
 See the [docs](https://github.com/MastodonC/witan.models.demography/blob/master/doc/intro.md) for more info about the methodology.
 
-## Development tools
+
+## Contribute to `witan.models.demography`
+
+### General practices
+
+If you wish to contribute to `witan.models.demography`, please follow read the following guidelines.
+
+#### Github guidelines
+* **Fork this repository** (or clone it if you have writing writes to the reposotory).
+* **Create a new branch**. Let's try and keep the following naming conventions for branches:
+  * `feature/<what-this-branch-solves>`, example: `feature/exception-logging`
+  * `doc/<what-this-branch-solves>`, example: `doc/contributors-practices`
+  * `fix/<what-this-branch-solves>`, example: `fix/run-ccm-from-cli`
+  * `tidy-up/<what-this-branch-solves>`, example: `tidy-up/upgrade-deps`
+
+  This way, when you see a branch starting by `fix/` we know something is broken and someone is repairing it.
+* **Keep branches short** so that the reviewing process is easier and faster
+* Start a pull request (PR) as earlier as possible. You can add a `WIP` in the title to specify it's in progress.
+* Describe the aim of your changes in the PR description box.
+* Before asking for a review of your PR:
+  * **run all the tests** from the commmand line with `$ lein test`
+  * **run the lint tools** [`Eastwood`](https://github.com/jonase/eastwood) and [`Kibit`](https://github.com/jonase/kibit) with `$ lein eastwood` and `$ lein kibit`.
+  * **format your code** with [`Cljfmt`](https://github.com/weavejester/cljfmt) tool with `lein cljfmt check` followed by `lein cljfmt fix`.
+  * **check the cohort-component model can still be run** from the command-line.
+  You'll need to build the jar file, run it on your terminal and check it outputs population projections.
+  For that follow the instructions in the ["user section"](#use-witanmodelsdemography) of the Readme.
+
+#### Coding guidelines
+
+* Write unit tests, docstrings, and documentation.
+* As much as possible try to not have data changes and code changes on the same branch or the same commit.
```

Comment:
maybe be clearer to say 'Try not to have data changes and code changes in the same commit, and preferably not the same branch, as the data tends to swamp the code and hinder reviewing'.